GpsDataAnalyzer/visualizer/visualizer.py

```diff
@@ -0,0 +1,139 @@
+"""
+Usage: visualizer.py
+
+Parse the xml or csv file and generate a GpsDataSet
+"""
+import matplotlib.pyplot as plt
+import matplotlib.dates as dates
+import numpy as np
+import seaborn as sns
+import pandas as pd
+import os
+
+from datetime import datetime
+from datetime import timedelta
+from datetime import timezone
+
+class Visualizer:
+    """
+    Classify the deivation of distance and visualize the deviations of distance/speed/altitude
+    """
+    def get_min_deviation(self, data):
+        """
+        Get the min value of deviation
+
+        Args:
+           data: the deviation data
+        """
+        return min(float(deviation) for deviation in data)
+
+    def get_max_deviation(self, data):
+        """
+        Get the max value of deviation
+
+        Args:
+           data: the deviation data
+        """
+        return max(float(deviation) for deviation in data)
+
+    def classify_deviation(self, deviation_dataframe):
+        """
+        Classify the deviation of distance according to its absolute value, and mark the data confidence (1, 2, 3). 
+        Higher score means higher confidence and acurancy.
+
+        Args:
+          deviation_dataframe: a dataframe constaining time and deviation of distance/speed/altitude 
+
+        Returns:
+          A dataframe with confidence
```

Comment:
`The same dataframe with the column "Confidence" added`?

Reply:
Yes, I should better phrase it.